hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSAppAttempt.java

```diff
@@ -1147,24 +1147,32 @@ private static boolean checkAndMarkRRVisited(
    * starvation.
    */
   List<ResourceRequest> getStarvedResourceRequests() {
+    // List of RRs we build in this method to return
     List<ResourceRequest> ret = new ArrayList<>();
+
+    // Track visited RRs to avoid the same RR at multiple locality levels
     Map<Priority, List<Resource>> visitedRRs= new HashMap<>();
 
+    // Start with current starvation and track the pending amount
     Resource pending = getStarvation();
     for (ResourceRequest rr : appSchedulingInfo.getAllResourceRequests()) {
       if (Resources.isNone(pending)) {
+        // Found enough RRs to match the starvation
         break;
       }
+
+      // See if we have already seen this RR
       if (checkAndMarkRRVisited(visitedRRs, rr)) {
         continue;
       }
 
-      // Compute the number of containers of this capability that fit in the
-      // pending amount
+      // A RR can have multiple containers of a capability. We need to
+      // compute the number of containers that fit in "pending".
       int ratio = (int) Math.floor(
```

Comment:
Given that ratio is the number of containers that fit in "pending," ratio is probably a bad name.  That was a good chunk of my initial confusion.